webapp/src/components/Utterance/UtteranceDataAction.tsx

```diff
@@ -24,9 +24,14 @@ const UtteranceDataAction: React.FC<Props> = ({
   const [updateDataAction] = updateDataActionsEndpoint.useMutation();
 
   const handleDataActionChange = (newValue: DataAction) => {
+    const utterancePatch: UtterancePatch[] = persistentIds.map(
+      (persistentId) => ({
+        persistentId,
+        dataAction: newValue,
+      })
+    );
     updateDataAction({
-      persistentIds,
-      newValue,
+      body: utterancePatch,
```

Comment:
I find it confusing that `utterancePatch` is in fact an array of `utterancePatch`es, plural. I think in this case I would simply call it body to avoid renaming it here.
```suggestion
      body,
```